### PR TITLE
[JENKINS-31206] Add checkbox to ArtifactDeployer to enable/disable POM upload with artifact

### PR DESCRIPTION
**Pull request by Brantone**
Had a scenario where uploading similar packages to Nexus while using "classifier" to differentiate.
Since this always deployed the POM file with it, it'd fail out. And since didn't even need the POM file, figured may as well remove it.
My circumstances are quite unique, but providing PR regardless.

### CODE REVIEW

File: src/main/java/org/jvnet/hudson/plugins/repositoryconnector/Artifact.java [hunk 1]
```diff
@@ -73,6 +75,17 @@ public String getTargetFileName() {
         return targetFileName;
     }
 
+    /**
+     * @return the uploadPOM
+     */
+    public boolean getUploadPOM() {
+        // Backwards compatibility: if null, config not updated: fallback to true
```
Comment (Brantone) [inline src/main/java/org/jvnet/hudson/plugins/repositoryconnector/Artifact.java]: This part here I was unsure of. Using "positive" booleans (ex: not have "disable = true") caused a bit of a backwards compatibility gotcha in terms of the value not being set in previous versions, reason for the class var being Boolean.
Open to suggestions on how to accomplish this more elegantly.

**Comment by imod** [comment]
I understand your usecase and I was bitten by the same issue before - but on the other hand, this is actually enforced by maven (or better the repository manager) and not following the rule will lead to inconsistent repositories. So I'm not sure we should really integrate this.

**Comment by Brantone** [comment]
That's assuming Maven is even in the picture ... because just about anything can be stored in Nexus (dmg, ipa, etc.), still requiring the POM ignores several other use cases.
That being said, recently a new plugin was created (https://wiki.jenkins-ci.org/display/JENKINS/Nexus+Artifact+Uploader) that I'm keeping my eye on and might swap for since it is specifically focussed on Nexus and like you said, this may be better suited with enforcement for Maven and avoid inconsistent repos. Could really go either way.

**Comment by jgangemi** [comment]
is there still a desire for this? right now poms are created per artifact but an "advance" option could be added that would allow for grouping multiple artifacts together under a single pom during deployment.

if yes, please file a jira issue.

closing due to #43 and not worth the effort of rebasing.

**Pull request closed without merging**